tcl/board.tcl

```diff
@@ -1542,6 +1569,13 @@ proc  ::board::lastMoveHighlight {w moveuci} {
     if { $::arrowLastMove } {
         ::board::mark::DrawArrow $w.bd $square1 $square2 $::highlightLastMoveColor
     }
+    set s ""
+    regexp \[!\?]+ $nag s
+    if { $s ne "" && $::highlightLastMoveNag  } {
```

Comment:
It is better to not use one char variables and the regexp should be executed only if the option is active:
if { $::highlightLastMoveNag && [regexp {[!?]+} $nag nag] } {